test/probot-octokit.test.ts

```diff
@@ -17,177 +17,346 @@ describe("ProbotOctokit", () => {
     },
   };
 
-  beforeEach(() => {
-    octokit = new ProbotOctokit(defaultOptions);
-  });
-
   test("works without options", async () => {
-    octokit = new ProbotOctokit();
-    const user = { login: "ohai" };
+    const fetch = fetchMock
+      .sandbox()
+      .getOnce("https://api.github.com/user", '{"login": "ohai"}');
 
-    nock("https://api.github.com").get("/user").reply(200, user);
-    expect((await octokit.users.getAuthenticated({})).data).toEqual(user);
+    const octokit = new ProbotOctokit({
+      ...defaultOptions,
+      request: {
+        fetch,
+      },
+    });
+
+    expect((await octokit.users.getAuthenticated({})).data).toEqual(
+      '{"login": "ohai"}'
+    );
   });
 
   test("logs request errors", async () => {
-    nock("https://api.github.com").get("/").reply(500, {});
+    const fetch = fetchMock.sandbox().getOnce("https://api.github.com/", {
+      status: 500,
+      body: {
+        message: "Internal Server Error",
+        documentation_url:
+          "https://docs.github.com/rest/reference/repos#get-repository-content",
+      },
+    });
+
+    const octokit = new ProbotOctokit({
+      ...defaultOptions,
+      request: {
+        fetch,
+      },
+    });
 
     try {
       await octokit.request("/");
       throw new Error("should throw");
     } catch (error) {
-      expect(error.status).toBe(500);
+      expect((error as RequestError).status).toBe(500);
     }
   });
 
-  describe("with retry enabled", () => {
-    beforeEach(() => {
-      const options: Options = {
-        ...defaultOptions,
-        retry: {
-          enabled: true,
-        },
-      };
-
-      octokit = new ProbotOctokit(options);
-    });
+  test("with retry enabled retries failed requests", async () => {
+    let callCount = 0;
 
-    test("retries failed requests", async () => {
-      nock("https://api.github.com").get("/").once().reply(500, {});
+    const octokit = new ProbotOctokit({
+      ...defaultOptions,
+      retry: {
+        enabled: true,
+      },
+      request: {
+        fetch: (url: string, options: { [key: string]: any }) => {
```

Comment:
i didnt know how could implement this with fetchMock. Any recommendations to do it in pure fetchMock?